File: api/models/comment.go
package models

import "gorm.io/gorm"

type Comment struct {
	gorm.Model
	Name    string `binding:"required" gorm:"type:varchar(255)"`
	Author  User   `binding:"required" gorm:"foreignKey:UserId"`
	UserId  uint64
	Movie   Movie `binding:"required" gorm:"foreignKey:MovieId"`
	MovieId uint64
	Rating  uint16 `binding:"required"`
}

File: frontendreact/movie_page/src/model/comment.ts
import { MovieModel } from "./movie";
import { UserModel } from "./user";

export interface CreateCommentModel {
    name: string | null,
    author: UserModel | null,
    movie: MovieModel | null,
    rating: number | null,
}

export interface ReadCommentModel {
    id: number | null,
    name: string | null,
    author: UserModel | null,
    createdDate: string | null,
    rating: number | null,
}